### PR TITLE
Added global switch to disable capturing stacks, and fixed typos in r…

### PR DESCRIPTION
**Pull request by ansel1**
…eadme.  @keeyim take a look.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,12 @@ always creates a new error which wraps the original.
  
 You can also add your own additional information.
 
+The stack capturing feature can be turned off for better performance, though it's pretty fast.  Benchmarks
```
Comment (keeyim) [inline README.md]: Nice, that's useful information.

File: README.md [hunk 2]
```diff
@@ -114,6 +120,8 @@ The package contains functions for creating new errors with stacks, or adding a
 instances.  Functions with add context (e.g. `WithValue()`) work on any `error`, and will 
 automatically convert them to merry errors (with a stack) if necessary.
 
+Capturing the stack can be globally disabled with `SetStackCaptureEnabled()`
```
Comment (keeyim) [inline README.md]: Would it be more clear here to say `SetStackCaptureEnabled(false)?`

File: errors_test.go [hunk 3]
```diff
@@ -411,3 +411,40 @@ func TestValues(t *testing.T) {
...
+	assert.NotEmpty(t, Stack(e))
+}
+
+func BenchmarkNew_withStackCapture(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		New("boom")
+	}
+}
+
+func BenchmarkNew_witoutStackCapture(b *testing.B) {
```
Comment (keeyim) [inline errors_test.go]: Nit: "without" vs "witout"

File: README.md [hunk 4]
```diff
@@ -28,6 +28,12 @@ always creates a new error which wraps the original.
  
 You can also add your own additional information.
 
+The stack capturing feature can be turned off for better performance, though it's pretty fast.  Benchmarks
+on an early 2011 MacBook Pro:
+
+    BenchmarkNew_withStackCapture-8     	 1000000	      1428 ns/op
```
Comment (keeyim) [inline README.md]: I assume the 1000000 here is the number of ops/loop size? Should that be clarified?

Also, what does the -8 in the name refer to?

File: README.md [hunk 5]
```diff
@@ -28,6 +28,12 @@ always creates a new error which wraps the original.
...
+    BenchmarkNew_witoutStackCapture-8   	10000000	       217 ns/op
```
Comment (keeyim) [inline README.md]: Why more iterations for without capture? Doesn't matter really, just curious.

**Comment by keeyim** [comment]
Ship it!

Thanks for adding this, I'll be using it right away. I may make it conditional: "debug" is on/default, non-"debug" is disabled.

**Comment by ansel1** [comment]
That benchmark output is what go test -bench spits out.  And it dictates the iterations.  It calls the routine enough times to get what it thinks is a stable average measurement.  That usually means, the faster the operation, the more iterations it uses to get a stable timing.

**Comment by keeyim** [comment]
I have an unrelated question. I’m not sure if there is a proper forum for asking such things on github; I was looking for a “Conversation” for the repository itself but couldn’t find such a thing. Please let me know if there is a better way to do this.

My question is: why did you choose to make the errors immutable?

I understand why one would not necessarily want to change the original, un-merry error. But for methods that are designed to operate on a merry error, why not modify the original error? I am not trying to say I think there is anything at all amiss here, I just want to better understand the rationale behind the design pattern.

**Comment by ansel1** [comment]
most repos handle questions with issues.  as you say, github doesn't offer anything forum-like built in yet.

The main reason for the design is that I was reading up on golang's `context` package at the time, and this was the design pattern it used.  The main advantage of it (that I cared about) is that its compatible with concurrency.  I don't expect merry errors to carry around more than 10 attributes, and for that, I suspect the performance and memory difference between a linked list and a map is negligible.

**Pull request merged**